main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
  4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
  */
 
+//UDT 1 // complete
 
-/*
-Thing 1) Laptop
-5 properties:
-    1) number of letter keys
-    2) number of numeric keys
-    3) amount of binary information in hard disk
-    4) number of usb ports
-    5) amount of electricity needed for portable use
-3 things it can do:
-    1) transfer information from usb device to hard disk
-    2) automatically adjust display brightness
-    3) provide update information to user
- */
 struct Laptop
 {
-	//number of letter keys
 	int numLetterKeys = 26;
-	//number of numeric keys
 	int numNumericKeys = 10;
-	//amount of binary information in hard disk
 	float binaryInfoInHardDisk = 499.96f;
-	//number of usb ports
 	int numUSBPorts = 2;
-	//amount of electricity needed for portable use
 	float electricityForPortableUse = 58.2f;
-	
-	struct User //first nested UDT
+
+	struct User
 	{
 		bool isUserMale = true;
 		int userAge = 37;
-		bool isUserAdmin = true;
+		bool isUserAdmin;
 		std::string userName = "Billy Crystal";
 		std::string password = "123456789";
+        int personalEmailTotal; // added member variable for nested 2nd implementation
+        int programmesOpen; // added member variable for nested 3rd implementation
```

Comment:
"for nested 2nd implementation"? 
what are you talking about?

Reply:
I made some new member variables so that the implementations below had something to return values to.

Reply:
like you "distanceTraveled" example